src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -301,6 +301,9 @@ ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
 	char relationKind = 0;
 	Var *distributionColumn = NULL;
 
+	/* we will not continue if there is a version mismatch */
+	ErrorIfInstalledVersionMismatch();
```

Comment:
If there is a version mismatch and the table is distributed, we error out at IsDistributedTable and IsDistributedTable covers most of the cases. But here we also should error out of table is not distributed.